### PR TITLE
Block emotes for sleeping

### PR DESCRIPTION
**Pull request by SkaldetSkaeg**
## About the PR

Blocked emotions while you sleep

## Why / Balance

Being able to scream while you sleep is very weird. It ruins the hypopen a bit. I don't have enough experience to rewrite such systems.

## Technical details

Just added cancel on EmoteAttemptEvent if entity is sleeping.
Ideally, we should add alternative emotions like "tossing and turning" or other sleepers things, but ChatSystem and VocalSystem are very outdated and have a lot of crutches.

## Requirements

- [X] I have read and am following the [Pull Request and Changelog Guidelines](https://docs.spacestation14.com/en/general-development/codebase-info/pull-request-guidelines.html).
- [X] I have added media to this PR or it does not require an ingame showcase.

## Breaking changes

**Changelog**

:cl:
- fix: Sleepers can no longer use emotions

### CODE REVIEW

**Comment by UbaserB** [comment]
this is more of a bugfix than a tweak i believe

File: Content.Shared/Bed/Sleep/SleepingSystem.cs [hunk 1]
```diff
@@ -310,6 +312,14 @@ public bool TryWaking(Entity<SleepingComponent?> ent, bool force = false, Entity
         Wake((ent, ent.Comp));
         return true;
     }
+
+    /// <summary>
+    /// Блокирует эмоции спящим
+    /// </summary>
```
Comment (metalgearsloth) [inline Content.Shared/Bed/Sleep/SleepingSystem.cs:318]: English please.

Comment (SkaldetSkaeg) [reply Content.Shared/Bed/Sleep/SleepingSystem.cs]: oopsie

File: Content.Shared/Bed/Sleep/SleepingSystem.cs [hunk 2]
```diff
@@ -310,6 +312,14 @@ public bool TryWaking(Entity<SleepingComponent?> ent, bool force = false, Entity
         Wake((ent, ent.Comp));
         return true;
     }
+
+    /// <summary>
+    /// Blocks emotions while sleeping
```
Comment (MilenVolf) [inline Content.Shared/Bed/Sleep/SleepingSystem.cs:317]:
```suggestion
    /// Prevents the use of emote actions while sleeping
```

**Comment by MilenVolf** [comment]
> this is more of a bugfix than a tweak i believe

Yeah, I also think that it's not the thing that players usually face with. So no CL needed I guess

**Comment by SkaldetSkaeg** [comment]
> > this is more of a bugfix than a tweak i believe
> 
> Yeah, I also think that it's not the thing that players usually face with. So no CL needed I guess

It was merged, so I have to somehow make these edits in a separate PR?

**Comment by UbaserB** [comment]
> It was merged, so I have to somehow make these edits in a separate PR?

its fine, not needed

**Pull request merged**